src/main/java/net/sf/jabref/importer/EntryFromPDFCreator.java

```diff
@@ -64,7 +64,7 @@ public boolean accept(File f) {
         PdfImporter pi = new PdfImporter(JabRefGUI.getMainFrame(), JabRefGUI.getMainFrame().getCurrentBasePanel(), JabRefGUI.getMainFrame().getCurrentBasePanel().getMainTable(), -1);
         ImportPdfFilesResult res = pi.importPdfFiles(Collections.singletonList(pdfFile.toString()));
         if (res.getEntries().size() == 1) {
-            return Optional.of(res.getEntries().get(0));
+            return Optional.ofNullable(res.getEntries().get(0));
```

Comment:
Bug fix so that one can click "Cancel" instead of selecting an entry type and still not get an NPE.